src/core_modules/capture-core/reducers/descriptions/currentSelections.reducerDescriptionGetter.js

```diff
@@ -47,21 +48,24 @@ const setCategoryOption = (
     };
 };
 
+const deleteKeyFromObject = (key, object) =>
+    Object.keys(object).reduce((acc, objectKey) => {
+        if (objectKey !== key) {
+            return { ...acc, [objectKey]: object };
+        }
+        return acc;
+    }, {});
+
 const resetCategoryOption = (state: Object, categoryId: string) => {
-    const categories = {
-        ...state.categories,
-        [categoryId]: undefined,
-    };
+    const { categoriesMeta, categories } = state;
 
-    const categoriesMeta = {
-        ...state.categoriesMeta,
-        [categoryId]: undefined,
-    };
+    const newCategories = deleteKeyFromObject(categoryId, categories);
+    const newCategoriesMeta = deleteKeyFromObject(categoryId, categoriesMeta);
 
     return {
         ...state,
-        categories,
-        categoriesMeta,
+        categories: newCategories,
+        categoriesMeta: newCategoriesMeta,
```

Comment:
Regarding my todo here. I was thinking, cant we simply have these two lines here be undefined?

What we do now is that each tome a user deselects a partner selection from the LockedSelector we remove that specific selection only from the object.

Reply:
There can be multiple category selections for a program (partner is just a preconfigured category in the demo db) so we need to keep this logic.

Reply:
ah right now I get it. makes sense